docs/end_to_end_tutorials/dev_practices/e2e_evaluation.md

```diff
@@ -0,0 +1,52 @@
+# End-to-End Evaluation
+End-to-End evaluation should be the guiding signal for your RAG application - will my pipeline generate the right responses given the data sources and a set of queries?
+
+While it helps initially to individually inspect queries and responses, as you deal with more failure and corner cases, it may stop being feasible to look at each query individually, and rather it may help instead to define a set of summary metrics or automated evaluation, and gain an intuition for what they might be telling you and where you might dive deeper.
+
+## Setting up an Evaluation Set
+
+It is helpful to start off with a small but diverse set of queries, and build up more examples as one discovers problematic queries or interactions.
+
+We've created some tools that automatically generate a dataset for you given a set of documents to query. (See example below).
+
+
+```{toctree}
+---
+maxdepth: 1
+---
+/examples/evaluation/QuestionGeneration.ipynb
+```
+
```

Comment:
after you setup a dataset, can we add a section on LLM-based eval? 

can link to our `ResponseEvaluator` and `QueryResponseEvaluator` modules here: https://gpt-index.readthedocs.io/en/latest/core_modules/supporting_modules/evaluation/modules.html - we evaluate query/context/responses in a label-free setting